signing_test.go

```diff
@@ -166,16 +172,39 @@ func TestSignerWithBrokenRand(t *testing.T) {
 
 	defer resetRandReader()
 
-	for _, alg := range sigAlgs {
+	for _, alg := range pssSigAlgs {
 		signingKey, verificationKey := GenerateSigningTestKey(alg)
-		for i, getReader := range readers {
+		for i, getReader := range brokenRandReaders {
 			RandReader = getReader()
 			err := RoundtripJWS(alg, serializer, corrupter, signingKey, verificationKey, "test_nonce")
 			if err == nil {
 				t.Error("signer should fail if rand is broken", alg, i)
 			}
 		}
 	}
+
+	// TODO(@pgporada) Remove this version checking
+	goVer, err := goVersionCheck()
+	if err != nil {
+		t.Error(err)
+	}
+	constraints, err := version.NewConstraint("< 1.20")
+	if err != nil {
+		t.Error("cannot setup version constraint")
+	}
+	if constraints.Check(goVer) {
+		pkcs1v15SigAlgs := []SignatureAlgorithm{RS256, RS384, RS512}
+		for _, alg := range pkcs1v15SigAlgs {
+			signingKey, verificationKey := GenerateSigningTestKey(alg)
+			for i, getReader := range brokenRandReaders {
+				RandReader = getReader()
+				err := RoundtripJWS(alg, serializer, corrupter, signingKey, verificationKey, "test_nonce")
+				if err == nil {
+					t.Error("signer should fail if rand is broken", alg, i)
+				}
+			}
```

Comment:
Rather than replicating this whole loop, do the version check higher up and then just conditionally append these three algorithms to the `[]SignatureAlgorithm`.